test/parallel_api/algorithm/alg.nonmodifying/nth_element.pass.cpp

```diff
@@ -122,17 +172,17 @@ test_by_type(Generator1 generator1, Generator2 generator2, Compare comp)
     for (size_t n = 0; n <= max_size; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
     {
         m = 0;
-        invoke_on_all_policies<0>()(test_one_policy<T>(), exp.begin(), exp.begin() + n, in1.begin(), in1.begin() + n,
-                                    n, m, generator1, generator2, comp);
+        test_multiple_interfaces<0, 1>()(exp.begin(), exp.begin() + n, in1.begin(), in1.begin() + n, n, m,
+                                        generator1, generator2, comp);
         m = n / 7;
-        invoke_on_all_policies<1>()(test_one_policy<T>(), exp.begin(), exp.begin() + n, in1.begin(), in1.begin() + n,
-                                    n, m, generator1, generator2, comp);
+        test_multiple_interfaces<2, 3>()(exp.begin(), exp.begin() + n, in1.begin(), in1.begin() + n, n, m,
+                                        generator1, generator2, comp);
         m = 3 * n / 5;
-        invoke_on_all_policies<2>()(test_one_policy<T>(), exp.begin(), exp.begin() + n, in1.begin(), in1.begin() + n,
-                                    n, m, generator1, generator2, comp);
+        test_multiple_interfaces<4, 5>()(exp.begin(), exp.begin() + n, in1.begin(), in1.begin() + n, n, m,
+                                        generator1, generator2, comp);
     }
-    invoke_on_all_policies<3>()(test_one_policy<T>(), exp.begin(), exp.begin() + max_size, in1.begin(),
-                                in1.begin() + max_size, max_size, max_size, generator1, generator2, comp);
+    test_multiple_interfaces<6, 7>()(exp.begin(), exp.begin() + max_size, in1.begin(), in1.begin() + max_size,
```

Comment:
Here you increase test cases twice: 8 instead of 4. Will it be better if we replace 2 of existing 4 with tests without predicate? In that case we will have 4 test cases, not 8

Reply:
(+) This will simplify the test: more readability and reliability
(-) This will reduce number of cases with predicate.

I would combine approaches by replacing one case within `for` loop and duplicating the last one.